polygeist/tools/cgeist/Lib/ValueCategory.cc

```diff
@@ -272,3 +272,13 @@ void ValueCategory::store(mlir::OpBuilder &builder, ValueCategory toStore,
     store(builder, toStore.getValue(builder));
   }
 }
+
+#define UNCONSTRAINED_CAST(FuncName, Op)                                       \
+  ValueCategory ValueCategory::FuncName(OpBuilder &Builder,                    \
+                                        Type PromotionType) const {            \
+    llvm::WithColor::warning() << "Creating unconstrained " #Op "\n";          \
+    return Cast<arith::Op>(Builder, PromotionType);                            \
+  }
+UNCONSTRAINED_CAST(FPTrunc, TruncFOp)
+UNCONSTRAINED_CAST(FPExt, ExtFOp)
+#undef UNCONSTRAINED_CAST
```

Comment:
Given these 2 functions are just 2 lines of code I would just define them directly rather than using a macro.

Reply:
There will be 4 more functions here (`FP->UI`, `FP->SI`, `SI->FP`, `UI->FP`). I can use a bridge function, though, which will look cleaner.